diffpy/snmf/io.py

```diff
@@ -0,0 +1,115 @@
+      parameter controlling sparseness of the solution, the matrix representing the smoothness term, and a matrix used
+      to construct a hessian matrix.
+
+    """
+    signal_length = data_input.shape[0]
+    moment_amount = data_input.shape[1]
+
+    component_matrix_guess = np.random.rand(signal_length, component_amount)
+    weight_matrix_guess = np.random.rand(component_amount, moment_amount)
+    stretching_matrix_guess = np.ones(component_amount, moment_amount) + np.random.randn(component_amount,
+                                                                                         moment_amount) * 1e-3
+
+    diagonals = [np.ones(moment_amount - 2), -2 * np.ones(moment_amount - 2), np.ones(moment_amount - 2)]
+    smoothness_term = .25 * scipy.sparse.diags(diagonals, [0, 1, 2], shape=(moment_amount - 2, moment_amount))
+
+    hessian_helper_matrix = scipy.sparse.block_diag([smoothness_term.T @ smoothness_term]*component_amount)
+    sequence = np.arange(moment_amount*component_amount).reshape(component_amount, moment_amount).T.flatten()
+    hessian_helper_matrix = hessian_helper_matrix[sequence, :][:, sequence]
+
+    return {
+        "signal_length": signal_length,
+        "moment_amount": moment_amount,
+        "component_matrix_guess": component_matrix_guess,
+        "weight_matrix_guess": weight_matrix_guess,
+        "stretching_matrix_guess": stretching_matrix_guess,
+        "component_amount": component_amount,
+        "data_type": data_type,
+        "smoothness": smoothness,
+        "sparsity": sparsity,
+        "smoothness_term": smoothness_term,
+        "hessian_helper_matrix": hessian_helper_matrix
+    }
+
+
+def load_input_signals(file_path=None):
+    """Processes a directory of a series of PDF/XRD patterns into a usable format.
+
+    Constructs a 2d array out of a directory of PDF/XRD patterns containing each files dependent variable column in a
+    new column. Constructs a 1d array containing the grid values.
+
+    Parameters
+    ----------
+    file_path: str or Path object, optional
+      The path to the directory containing the input XRD/PDF data. If no path is specified, defaults to the current
+      working directory. Accepts a string or a pathlib.Path object. Input data not on the same grid as the first file
+      read will be ignored.
+
+    Returns
+    -------
+    tuple
+      The tuple whose first element is an R x M 2d array made of PDF/XRD patterns as each column; R is the length of the
+      signal and M is the number of patterns. The tuple contains a 1d array containing the values of the grid points as
+      its second element; Has length R.
+
+    """
+
+    if file_path is None:
+        directory_path = Path.cwd()
+    else:
+        directory_path = Path(file_path)
+
+    values_list = []
+    grid_list = []
+    current_grid = []
+    for item in directory_path.iterdir():
+        if item.is_file():
```

Comment:
this may be ok, but I am more used to seeing file handling being carried out in a context manager context which is considered to be more safe.  There may be some nasty unwanted side-effects here, or is that handled inside `loadData`?  Probably good to check.

Reply:
I will look into loadData and get back to you. To make sure, by 'context manager' do you mean file handling using things such as with statement or classes?

Reply:
maybe google and read about python context managers?

Reply:
It seems that loadData does file handling in a context manager context.